reasoner/processor/AbstractProcessor.java

```diff
@@ -182,6 +185,36 @@ public Driver<Monitor> monitor() {
             return monitor;
         }
 
+        public ReasonerPerfCounters perfCounters() {
+            return perfCounters;
+        }
+
+        public static class ReasonerPerfCounters {
+
+            public static final String KEY_TIME_PLANNING_MS = "time_planning_ms";
+            public static final String KEY_COUNT_MATERIALISATIONS = "count_materialisations";
+            public static final String KEY_COUNT_CONJUNCTION_PROCESSORS = "count_conjunction_processors";
+            public static final String KEY_COUNT_COMPOUND_STREAMS = "count_compound_streams";
+
+            private final PerfCounters perfCounters;
+
+            public final PerfCounters.Counter timePlanningMs;
+            public final PerfCounters.Counter countMaterialisations;
+            public final PerfCounters.Counter countConjunctionProcessors;
+            public final PerfCounters.Counter countCompoundStreams;
+
+            public ReasonerPerfCounters(boolean enabled) {
+                perfCounters = new PerfCounters(enabled);
+                timePlanningMs = perfCounters.register(KEY_TIME_PLANNING_MS);
+                countMaterialisations = perfCounters.register(KEY_COUNT_MATERIALISATIONS);
+                countConjunctionProcessors = perfCounters.register(KEY_COUNT_CONJUNCTION_PROCESSORS);
+                countCompoundStreams = perfCounters.register(KEY_COUNT_COMPOUND_STREAMS);
+            }
+
+            public Map<String, Long> snapshotUnsynchronised() {
+                return perfCounters.snapshotUnsynchronised();
+            }
```

Comment:
I've changed my mind - think this class should actually extend the common `PerfCounter` class. Sorry! 

Then we can get rid of the duplication and the awkwardness that a `PerfCounters` holds onto a `PerfCounters` object. However, if it was initialised with a set of `Counter` directly (eg. doesn't create a `new PerfCounters()` itself) then I would still think it is better not extending `PerfCounters`

Reply:
I'm currently creating ReasonerPerfCounters as an extension of PerfCounters, but we can remove the PerfCounter*s* class and just keep the counters themselves if we don't see them as a tightly grouped set.